R/mkt_analysis.R

```diff
@@ -0,0 +1,78 @@
+#' @title Find percent of markets that are a desert for a specific type of desert
+#' @export
+pct_desert <- function(df) {
+
+  df %>%
+    dplyr::group_by(desert_type, desert) %>%
+    dplyr::summarise(n = dplyr::n()) %>% 
+    dplyr::mutate(pct = round((n/sum(n))*100,1)) %>%
+    dplyr::filter(desert) %>%
+    dplyr::select(-n) %>%
+    tidyr::pivot_wider(names_from = desert_type, values_from = pct)
+}
+
+#' @title Find total number of children in need of care of each type
+#' @export
+total_children <- function(df) {
+  
+  df %>%
+    dplyr::group_by(desert) %>%
+    dplyr::summarise(n_kids = round(sum(tract_demand, na.rm = T))) %>%
+    tidyr::pivot_wider(names_from = desert, values_from = n_kids)
+}
+
+#' @title Find total number of seats of each type
+#' @export
+total_seats <- function(df){
+
+  df %>%
+    dplyr::group_by(desert) %>%
+    dplyr::summarise(n_seats = round(sum(tract_supply, na.rm = T))) %>%
+    tidyr::pivot_wider(names_from = desert, values_from = n_seats)
+}
+
+#' @title Find average number of children per market
+#' @export
+avg_children_mkt <- function(df) {
+
+  df %>%
+    dplyr::group_by(desert_type) %>%
+    dplyr::summarise(n_kids = round(mean(mkt_demand, na.rm = T))) %>%
+    tidyr::pivot_wider(names_from = desert_type, values_from = n_kids)
+}
+
+#' @title Find average number of seats per market
+#' @export
+avg_seats_mkt <- function(df){
+
+  df %>%
+    dplyr::group_by(desert_type) %>%
+    dplyr::summarise(n_seats = round(mean(mkt_supply, na.rm = T))) %>%
+    tidyr::pivot_wider(names_from = desert_type, values_from = n_seats)
+}
+
+#' @title Find average number of providers per market
+#' @export
+avg_provider_mkt <- function(xwalk_tract_provider) {
+
+  xwalk_tract_provider %>%
+    dplyr::group_by(anchor_tract) %>%
+    dplyr::summarise(n_provider = dplyr::n_distinct(operation_number)) %>%
+    dplyr::summarise(n_prvdr = mean(n_provider))
+}
+
+#' @title Find number of children per desert type
+#' @export
+total_children_desert <- function(df_ratio,
+                                  df_demand) {
+
+  df_ratio %>%
+    dplyr::filter(desert) %>%
```

Comment:
@sadierichardson this is the one substantial change I made to the code. I removed an inner join, which I think was double-counting some of the kids. I was getting really high percentages like ~91% of kids live in a TRS desert, which didn't seem right. I think we just want filter to deserts by desert type and sum the number of kids in each anchor tract. If we add in the kids from the surrounding tracts then they can be counted twice which inflates our numbers. This isn't a perfect system, but it's the methodology we came up with before.

Reply:
@holmesjoli That makes sense. I sort of struggled with how we wanted to do this and ended up changing the method for the plots but not for this part. I think counting the kids per desert is kind of misleading in general because of the nature of the markets and kids belonging to multiple... but if this is how we did before, then it's probably good enough.

Reply:
Yes, I brought up the exact same comment last year. I completely agree with you, but this was how Christina wanted to do it!

Reply:
@holmesjoli Sounds good!